spec/unit/lib/use_cases/update_rent_charges_spec.rb

```diff
@@ -0,0 +1,113 @@
+    update_rent_charges.execute({ year: 2016, rc_uplift: -0.01 })
+    expect(rent_charges_gateway).to have_received(:save) do |rent_charges|
+      expect(rent_charges[0][:uprn]).to eq(rent_charges_for_given_year[0][:uprn])
+      expect(rent_charges[0][:address]).to eq(rent_charges_for_given_year[0][:address])
+      expect(rent_charges[0][:address_2]).to eq(rent_charges_for_given_year[0][:address_2])
+      expect(rent_charges[0][:comments]).to eq(rent_charges_for_given_year[0][:comments])
+      expect(rent_charges[0][:rr_count]).to eq(1)
+      expect(rent_charges[0][:property_type]).to eq(rent_charges_for_given_year[0][:property_type])
+      expect(rent_charges[0][:base_data_bed_size]).to eq(rent_charges_for_given_year[0][:base_data_bed_size])
+      expect(rent_charges[0][:bedroom_weight]).to eq(rent_charges_for_given_year[0][:bedroom_weight])
+      expect(rent_charges[0][:mra_archetype]).to eq(rent_charges_for_given_year[0][:mra_archetype])
+      expect(rent_charges[0][:jan_1999_asset_values]).to eq(rent_charges_for_given_year[0][:jan_1999_asset_values])
+      expect(rent_charges[0][:year]).to eq(2016)
+      expect(rent_charges[0][:formula_rent_this_year]).to eq((rent_charges_for_given_year[0][:formula_rent_this_year] * 0.99).round(2))
+      expect(rent_charges[0][:rent_cap_this_year]).to eq((rent_charges_for_given_year[0][:rent_cap_this_year]. * 0.99).round(2))
+      expect(rent_charges[0][:uprated_actual]).to eq((rent_charges_for_given_year[0][:uprated_actual].round(2) * 0.99).round(2))
+
+      expect(rent_charges[1][:uprn]).to eq(rent_charges_for_given_year[1][:uprn])
+      expect(rent_charges[1][:address]).to eq(rent_charges_for_given_year[1][:address])
+      expect(rent_charges[1][:address_2]).to eq(rent_charges_for_given_year[1][:address_2])
+      expect(rent_charges[1][:comments]).to eq(rent_charges_for_given_year[1][:comments])
+      expect(rent_charges[1][:rr_count]).to eq(1)
+      expect(rent_charges[1][:property_type]).to eq(rent_charges_for_given_year[1][:property_type])
+      expect(rent_charges[1][:base_data_bed_size]).to eq(rent_charges_for_given_year[1][:base_data_bed_size])
+      expect(rent_charges[1][:bedroom_weight]).to eq(rent_charges_for_given_year[1][:bedroom_weight])
+      expect(rent_charges[1][:mra_archetype]).to eq(rent_charges_for_given_year[1][:mra_archetype])
+      expect(rent_charges[1][:jan_1999_asset_values]).to eq(rent_charges_for_given_year[1][:jan_1999_asset_values])
+      expect(rent_charges[1][:year]).to eq(2016)
+      expect(rent_charges[1][:formula_rent_this_year]).to eq((rent_charges_for_given_year[1][:formula_rent_this_year] * 0.99).round(2))
+      expect(rent_charges[1][:rent_cap_this_year]).to eq((rent_charges_for_given_year[1][:rent_cap_this_year]. * 0.99).round(2))
+      expect(rent_charges[1][:uprated_actual]).to eq((rent_charges_for_given_year[1][:uprated_actual].round(2) * 0.99).round(2))
+    end
+  end
+
+  it 'raises error when missing year' do
```

Comment:
Might use terminology like `returns an error` or `includes an error` rather than `raises an error`. Super nitpicky, but `raise` has very specific connotations.